src/components/pin/pin.repository.ts

```diff
@@ -22,28 +22,29 @@ export class PinRepository {
     return result ? true : false;
   }
 
-  async togglePinned(
-    id: string,
-    pinned: boolean,
-    session: Session
-  ): Promise<void> {
-    await this.removeOldPinned(id, session);
-    if (pinned) {
-      const createdAt = DateTime.local();
-      await this.db
-        .query()
-        .call(matchRequestingUser, session)
-        .match([node('node', 'BaseNode', { id })])
-        .create([
-          node('requestingUser'),
-          relation('out', '', 'pinned', { active: true, createdAt }),
-          node('node'),
-        ])
-        .run();
+  async add(id: string, session: Session): Promise<void> {
+    const isPinned = await this.isPinned(id, session);
+    if (isPinned) {
+      return;
     }
+    const createdAt = DateTime.local();
+    await this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'BaseNode', { id })])
+      .create([
+        node('requestingUser'),
+        relation('out', '', 'pinned', { active: true, createdAt }),
+        node('node'),
+      ])
+      .run();
   }
 
-  protected async removeOldPinned(id: string, session: Session): Promise<void> {
+  async remove(id: string, session: Session): Promise<void> {
+    const isPinned = await this.isPinned(id, session);
+    if (!isPinned) {
+      return;
+    }
```

Comment:
Repositories should just contain low level db queries. They should not be smart, which is why they are private to the module. All business-logic / smarts should go in the service.

It also looks like it's ok to call the query below even if the user hasn't pinned this right?

Reply:
yeah, but there will be duplicated relations.

Reply:
How? It's just deleting if found right?